### PR TITLE
[dart-dio] [dart] Unify both dart generators into just `dart`

### PR DESCRIPTION
**Pull request by ahmednfwela**
based on: https://github.com/OpenAPITools/openapi-generator/pull/14291
Blocked on: 
- https://github.com/OpenAPITools/openapi-generator/pull/15485
- https://github.com/OpenAPITools/openapi-generator/issues/15449

this PR adds support for using http library instead of dio

it's still WIP and highly experimental, suggestions welcome

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: bin/configs/dart-dio-echo-server-dio-built_value.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+templateDir: modules/openapi-generator/src/main/resources/dart/libraries/dio
+typeMappings:
+  Client: "ModelClient"
+  File: "ModelFile"
+  EnumClass: "ModelEnumClass"
+additionalProperties:
+  hideGenerationTimestamp: "true"
+  enumUnknownDefaultCase: "true"
+  serializationLibrary: "built_value"
+  networkingLibrary: "dio"
```
Comment (wing328) [inline bin/configs/dart-dio-echo-server-dio-built_value.yaml:13]: What about using `library` (global option, not generator option) instead?

Comment (ahmednfwela) [reply bin/configs/dart-dio-echo-server-dio-built_value.yaml:13]: currently the `serializationLibrary` is assigned to `library` global option (this was introduced in https://github.com/OpenAPITools/openapi-generator/pull/9980)
I was going to make another PR that targets v7.0.0, with the correct parameters assigned, but I am not sure if this is the right approach

Comment (ahmednfwela) [reply bin/configs/dart-dio-echo-server-dio-built_value.yaml:13]: relevant line: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java#L162

**Comment by ahmednfwela** [comment]
Replaced by https://github.com/OpenAPITools/openapi-generator/pull/15485

**Pull request closed without merging**